components/applications-service/pkg/server/periodic.go

```diff
@@ -249,3 +392,50 @@ func (m *MarkDisconnectedServicesExecutor) runWithoutStats(t cereal.Task) error
 func (m *MarkDisconnectedServicesExecutor) TotalRuns() int64 {
 	return atomic.LoadInt64(&m.totalRuns)
 }
+
+type DeleteDisconnectedServicesExecutor struct {
+	ApplicationsServer  *ApplicationsServer
+	totalRuns           int64
+	totalRunsFailed     int64
+	totalRunsSuccessful int64
+}
+
+func (d *DeleteDisconnectedServicesExecutor) Run(ctx context.Context, t cereal.Task) (interface{}, error) {
+	err := d.runWithoutStats(ctx, t)
+	atomic.AddInt64(&d.totalRuns, 1)
+	if err != nil {
+		atomic.AddInt64(&d.totalRunsFailed, 1)
+	} else {
+		atomic.AddInt64(&d.totalRunsSuccessful, 1)
+	}
+	return nil, err
+}
+
+func (d *DeleteDisconnectedServicesExecutor) runWithoutStats(ctx context.Context, t cereal.Task) error {
```

Comment:
I'm not sure I understand what `withoutStats` implies here

Reply:
Looks like it just doesn't increment runs succeeded and failed.

Reply:
The one caller this method has is `Run` that just calls this and manages the stats/counters... I did it that way so there's only one place that has to update the counter for failed runs instead of repeating that in each err-early-return clause.